src/model/wallet.js

```diff
@@ -298,6 +566,124 @@ const loadPrivateKey = async (
   }
 };
 
-export const saveAddress = async (address, accessControlOptions = {}) => {
-  await keychain.saveString(addressKey, address, accessControlOptions);
+export const generateAccount = async (id, index) => {
+  try {
+    const authenticationPrompt = lang.t('wallet.authenticate.please');
+    const isSeedPhraseMigrated = await keychain.loadString(
+      seedPhraseMigratedKey,
+      {
+        authenticationPrompt,
+      }
+    );
+    let seedPhrase, hdnode;
+    // We need to migrate the seedphrase & private key first
+    // In that case we regenerate the existing private key to store it with the new format
+    if (!isSeedPhraseMigrated) {
+      const {
+        hdnode: newHdnode,
+        seedPhrase: newSeedPhrase,
+      } = await migrateSecrets();
+
+      if (newHdnode) {
+        hdnode = newHdnode;
+      }
+
+      if (newSeedPhrase) {
+        seedPhrase = newSeedPhrase;
+      }
+    } else {
+      const seedData = await getSeedPhrase(id);
+      seedPhrase = seedData.seedphrase;
+      hdnode = ethers.utils.HDNode.fromMnemonic(seedPhrase);
+    }
+
+    if (!seedPhrase) {
+      throw new Error(`Can't access seed phrase to create new accounts`);
+    }
+
+    const node = hdnode.derivePath(`${DEFAULT_HD_PATH}/${index}`);
+    const newAccount = new ethers.Wallet(node.privateKey);
+    await savePrivateKey(newAccount.address, newAccount.privateKey);
+    return newAccount;
+  } catch (error) {
+    logger.log('Error generating account for keychain', id, error);
+  }
+};
+
+export const migrateSecrets = async () => {
+  try {
+    const publicAccessControlOptions = {
+      accessible: ACCESSIBLE.ALWAYS_THIS_DEVICE_ONLY,
+    };
+    const seedPhrase = await loadSeedPhrase();
```

Comment:
can we add something to the function name `loadSeedPhrase` that makes it obvious that it was part of the old version